etl/steps/data/garden/war/2023-07-07/mars.py

```diff
@@ -0,0 +1,298 @@
+    - We encode the conflict type in two categories: 'civil war' and 'others (non-civil war)'.
+    """
+    ## Get region label
+    tb = tb.melt(id_vars=COLUMNS_CODES + COLUMNS_YEARS + COLUMNS_METRICS + [COLUMN_CIVIL_WAR], var_name="region")
+    tb = tb[tb["value"] == 1].drop(columns="value")
+
+    ## Get conflict_type info
+    tb[COLUMN_CIVIL_WAR] = tb[COLUMN_CIVIL_WAR].map({0: "others (non-civil)", 1: "civil war"})
+    tb = tb.rename(columns={COLUMN_CIVIL_WAR: "conflict_type"})
+
+    return tb
+
+
+def reduce_triplets(tb: Table) -> Table:
+    """Reduce table to have only one row per (warcode, campcode, ccode) triplet.
+
+    - There are few instances where there are duplicate entries with the same triplet.
+    - This function aggregates the number of deaths for these, and takes the start and end year of the triplet as the min and max, respectively.
+    - If duplicated triplets present different regions or conflict types, this will raise an error!
+    """
+    ## Combine duplicated tripplets ("warcode", "campcode", "ccode")
+    tb = tb.groupby(["warcode", "campcode", "ccode"], as_index=False).agg(
+        {
+            "yrstart": "min",
+            "yrend": "max",
+            "kialow": "sum",
+            "kiahigh": "sum",
+            "region": lambda x: list(set(x))[0] if len(set(x)) == 1 else np.nan,
+            "conflict_type": lambda x: list(set(x))[0] if len(set(x)) == 1 else np.nan,
+        }
+    )
+    assert tb.isna().sum().sum() == 0, "Unexpected NaNs were found!"
+
+    return tb
+
+
+def expand_observations(tb: Table) -> Table:
+    """Add entries to each triplet for all its years of activity.
+
+    - By default, a triplet only appears in a row, with the total number of fatalities and the year of start and end.
+    - This function expands these single entries in order to obtain a row per triplet + year of occurence.
+    - Deaths per year are estimated assuming uniform distribution.
+
+    Example
+
+        Input:
+
+        | warcode | campcode | ccode | year_start | year_end | deaths |
+        |---------|----------|-------|------------|----------|--------|
+        | 1       | 1        | 1     | 1990       | 1993     | 12     |
+
+        Output:
+
+        |  year | warcode | campcode | ccode | deaths |
+        |-------|---------|----------|-------|--------|
+        |  1990 |    1    |    1     | 1     |   3    |
+        |  1991 |    1    |    1     | 1     |   3    |
+        |  1992 |    1    |    1     | 1     |   3    |
+        |  1993 |    1    |    1     | 1     |   3    |
+
+    """
+    # Add an entry of the triplet ("warcode", "campcode", "ccode") per year
+    # For that we scale the number of deaths proportional to the duration of the conflict.
+    tb[["kialow", "kiahigh"]] = tb[["kialow", "kiahigh"]].div(tb["yrend"] - tb["yrstart"] + 1, "index")
```

Comment:
What happens with non-integer deaths here?

Reply:
In line 79, we assign type `Int64` to these columns. Although this converts floats to integers, it applies always `floor` operation.

Instead I'll add a `round` call, so that actually a rounding operation is performed.